cis194/week4/zehua/src/Exercises.hs

```diff
@@ -0,0 +1,27 @@
+module Exercises where
+
+import Data.Bool (bool)
+
+-- ex1
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
+  | even x = (x - 2) * fun1 xs
+  | otherwise = fun1 xs
+
+fun1' :: [Integer] -> Integer
+fun1' = foldr ((*) . (-2 +)) 1 . filter even
```

Comment:
```suggestion
fun1' = product . (subtract 2) . filter even
```

Reply:
do you mean `map (subtract 2)`

Reply:
> do you mean `map (subtract 2)`

Yep, `product . fmap (subtract 2) . filter even`.
Sorry for the confusion...